synapse/storage/data_stores/main/events.py

```diff
@@ -1053,6 +1053,13 @@ def _censor_redactions(self):
         if self.hs.config.redaction_retention_period is None:
             return
 
+        if self.db.updates.has_completed_background_update(
```

Comment:
async function is async

Reply:
Yup, so I noticed this, stuck a yield in, and then started questioning yielding on coroutine and :sob: :sob: :sob:

Reply:
Also, there should be a `not` there.